packages/wonder-blocks-form/src/components/__tests__/text-area.test.tsx

```diff
@@ -1324,5 +1351,415 @@ describe("TextArea", () => {
+                    ["t"],
+                    ["te"],
+                    ["tes"],
+                    ["test"],
+                ]);
+            });
+
+            describe("instantValidation=true", () => {
+                it("should call validate each time the value changes if the instantValidation prop is true", async () => {
+                    // Arrange
+                    const validate = jest.fn();
+                    render(
+                        <ControlledTextArea
+                            validate={validate}
+                            instantValidation={true}
+                        />,
+                        defaultOptions,
+                    );
+
+                    // Act
+                    const field = screen.getByRole("textbox");
+                    await userEvent.type(field, "test");
+                    await userEvent.tab();
+
+                    // Assert
+                    expect(validate.mock.calls).toStrictEqual([
+                        ["t"],
+                        ["te"],
+                        ["tes"],
+                        ["test"],
+                    ]);
+                });
+
+                it("should call onValidate with the error message each time the value changes", async () => {
+                    // Arrange
+                    const onValidate = jest.fn();
+                    const errorMessage = "Error";
+                    render(
+                        <ControlledTextArea
+                            validate={() => errorMessage}
+                            onValidate={onValidate}
+                            instantValidation={true}
+                        />,
+                        defaultOptions,
+                    );
+
+                    // Act
+                    const field = screen.getByRole("textbox");
+                    await userEvent.type(field, "test");
+                    await userEvent.tab();
+
+                    // Assert
+                    expect(onValidate.mock.calls).toStrictEqual([
+                        [errorMessage],
+                        [errorMessage],
+                        [errorMessage],
+                        [errorMessage],
+                    ]);
+                });
+
+                it("should have the textarea in an error state after validation fails without waiting for the user to tab away", async () => {
+                    // Arrange
+                    render(
+                        <ControlledTextArea
+                            instantValidation={true}
+                            validate={() => "Error message"}
+                        />,
+                        defaultOptions,
+                    );
+                    // Act
+                    const field = screen.getByRole("textbox");
+                    await userEvent.type(field, "test");
+
+                    // Assert
+                    expect(field).toHaveAttribute("aria-invalid", "true");
+                });
```

Comment:
Clarification from our discussion offline: This refers to the case where the field is empty, the user focuses on it, and verify that `validate` is not called when instantValidation=true

As discussed, this case is unrelated to the `instantValidation` prop since validating on mount depends on if there is a value set in the component! It is covered in these tests:
https://github.com/Khan/wonder-blocks/blob/d2aeaf80f44355651f0e5662ad93ba4944fa45c0/packages/wonder-blocks-form/src/components/__tests__/text-area.test.tsx#L980-L1012

Let me know if this addresses what you had in mind! :)

Reply:
thanks for the link, the second test covers the case I was thinking about.